### ISSUE TITLE
Odd phenomenon where training loss steps down after evaluation

### ISSUE THREAD

**Issue by alekseynp** [facebookresearch/detectron2#1950]
Sorry for the half-baked issue. I promise to turn this into a more complete issue once I get to the bottom of it.

I observe the following strange phenomenon that I seek an explanation for.

Here are two runs training loss visualized in tensorboard.
The dark red line steps down LR at 17.5k and 22.5k iterations, the blue line does not. This explains the overall shape of the curves. However...
My EVAL_PERIOD is 1,260 steps and I observe a distinct step pattern at exactly that interval in the training loss. This is highly unexpected.

![image](https://user-images.githubusercontent.com/4382608/91001311-3d957100-e580-11ea-90e2-5fb1fd3e65cf.png)

I'm looking for hypotheses and digging into details here. What about the evaluation phase timing could affect train loss?

Could it be that evaluation is happening with the model in training mode and the batch norm layers are "learning" during this phase? It seems not.
Are data loaders potentially being re-created? Is the model being served the exact same batches after each evaluation? I'm not sure.
Other ideas?

Thanks for your attention. I hope this isn't caused by the specifics of my setup, but is rather something fundamental in the detectron2 training regime that I can help identify

**Comment by ppwwyyxx** [comment]
The period in the curve seems to be around 2.5k so not exactly EVAL_PERIOD. Could it be related to period in a custom data loader?
The default training data loaders are not recreated and the default models should have handled batchnorm correctly.

**Comment by alekseynp** [comment]
So you're right, that periodicity is exactly my epochs.
40,000 images, 20 ims per batch, 2,500 batches.
Using detectron2.engine.launch
1 machine, 2 gpus
Trainer based on DefaultTrainer with no custom data loaders
I am not grouping by aspect ratio

I hacked in a dump of image_ids by rank and batch_id, and I am satisfied that they are shuffling properly.

**Comment by ppwwyyxx** [comment]
Then it does not sound like an unexpected issue (that evaluation could affect training) to me. Period in training data can affect training curves and we have observed such pattern as well.
